File: tasks/translation/output/Qwen/Qwen2.5-Coder-32B-Instruct-AWQ/codenet/cot/Java/Python/s527301188_commented.py
import sys

input = sys.stdin.read().strip()
strs = input.split()
nums = []

for s in strs:
    if s == '+':
        nums[-2] = nums[-2] + nums[-1]
        nums.pop()
    elif s == '-':
        nums[-2] = nums[-2] - nums[-1]
        nums.pop()
    elif s == '*':
        nums[-2] = nums[-2] * nums[-1]
        nums.pop()
    else:
        nums.append(int(s))

print(nums[0])
# End of Code

File: tasks/translation/output/Qwen/Qwen2.5-Coder-32B-Instruct-AWQ/codenet/cot/Java/C++/s527301188_commented.cpp
#include <iostream>
#include <vector>
#include <string>
#include <sstream>

int main() {
    std::string input;
    std::getline(std::cin, input);
    std::istringstream iss(input);
    std::vector<int> nums;
    std::string token;

    while (iss >> token) {
        switch (token[0]) {
            case '+':
                nums[nums.size() - 2] = nums[nums.size() - 2] + nums[nums.size() - 1];
                nums.pop_back();
                break;
            case '-':
                nums[nums.size() - 2] = nums[nums.size() - 2] - nums[nums.size() - 1];
                nums.pop_back();
                break;
            case '*':
                nums[nums.size() - 2] = nums[nums.size() - 2] * nums[nums.size() - 1];
                nums.pop_back();
                break;
            default:
                nums.push_back(std::stoi(token));
                break;
        }
    }
    std::cout << nums[0] << std::endl;

    return 0;
}
// End of Code
